app/controllers/users/sessions_controller.rb

```diff
@@ -217,6 +221,7 @@ def track_authentication_attempt
       success = current_user.present? &&
                 !user_locked_out?(user) &&
                 (recaptcha_response.success? || log_captcha_failures_only?)
+      session[:recaptcha_assessment_id] = recaptcha_assessment_id if recaptcha_assessment_id
```

Comment:
To leave open the option of using reCAPTCHA for other, non-sign-in features, I think it could be wise to add a qualifier here, similar to what we do with `user_session[:phone_recaptcha_assessment_id]` (e.g. `session[:sign_in_recaptcha_assessment_id]`).

Reply:
This was my open question #3: 
> Use plain :recaptcha_assessment_id in session as there's no value in a more specific namespace

Will Google assign different assessment ids for the same device session?

My thought was to create a ticket to move `user_session[:phone_recaptcha_assessment_id]` also to use `session[:recaptcha_assessment_id]`.

Reply:
They're separate assessments, and we'd want to annotate them separately. I don't think we want to converge the values.